pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -155,97 +160,107 @@ func (s *statesInformer) reportNodeTopology() {
 	nodeName := s.node.Name
 	s.nodeRWMutex.RUnlock()
 	ctx := context.TODO()
-	cpuTopology, usedCPUs, err := s.calCpuTopology()
+	nodeCPUInfo, cpuTopology, sharedPoolCPUs, err := s.calCPUTopology()
 	if err != nil {
 		return
 	}
 
-	// default policy is none
-	cpuManagerPolicy := extension.KubeletCPUManagerPolicy{
-		Policy: "none",
-	}
 	kubeletPort := int(s.GetNode().Status.DaemonEndpoints.KubeletEndpoint.Port)
-	filePath, err := system.GuessConfigFilePathFromKubeletPort(kubeletPort)
+	args, err := getKubeletCommandlineFn(kubeletPort)
 	if err != nil {
-		klog.Error("failed to read config file path from kubelet args")
-	}
-	if filePath != "" {
-		kubeletConfig, err := loadConfigFile(filePath)
-		if err != nil {
-			klog.Errorf("failed to load kubelet config, err: %v", err)
-		} else {
-			if len(kubeletConfig.CPUManagerPolicyOptions) != 0 {
-				cpuManagerPolicy.Options = kubeletConfig.CPUManagerPolicyOptions
-			}
-			if kubeletConfig.CPUManagerPolicy != "" {
-				cpuManagerPolicy.Policy = kubeletConfig.CPUManagerPolicy
-			}
-		}
+		klog.Errorf("Failed to GetKubeletCommandline with kubeletPort %d, err: %v", kubeletPort, err)
+		return
 	}
 
-	cpuPolicy, stateFilePath, cpuManagerOpt, err := system.GuessCPUManagerOptFromKubeletPort(kubeletPort)
+	klog.V(5).Infof("kubelet args: %v", args)
+
+	kubeletOptions, err := kubeletutil.NewKubeletOptions(args)
 	if err != nil {
-		klog.Errorf("failed to guess kubelet cpu manager opt, err: %v", err)
+		klog.Errorf("Failed to NewKubeletOptions, err: %v", err)
+		return
 	}
-	// use args first
-	if cpuPolicy != "" {
-		cpuManagerPolicy.Policy = cpuPolicy
+
+	// default policy is none
+	cpuManagerPolicy := extension.KubeletCPUManagerPolicy{
+		Policy:  kubeletOptions.CPUManagerPolicy,
+		Options: kubeletOptions.CPUManagerPolicyOptions,
 	}
-	if len(cpuManagerOpt) != 0 {
-		cpuManagerPolicy.Options = cpuManagerOpt
+
+	if kubeletOptions.CPUManagerPolicy == string(cpumanager.PolicyStatic) {
+		topology := kubeletutil.NewCPUTopology((*util.LocalCPUInfo)(nodeCPUInfo))
+		reservedCPUs, err := kubeletutil.GetStaticCPUManagerPolicyReservedCPUs(topology, kubeletOptions)
+		if err != nil {
+			klog.Errorf("Failed to GetStaticCPUManagerPolicyReservedCPUs, err: %v", err)
+		}
+		cpuManagerPolicy.ReservedCPUs = reservedCPUs.String()
+
+		for _, cpuID := range reservedCPUs.ToSliceNoSort() {
```

Comment:
as we discussed keep reserved cpu inside share pool for burstable pod(req=0, limit=1)